include/boost/url/params_encoded.hpp

```diff
@@ -49,11 +49,21 @@ class params_encoded
         url& u) noexcept;
 
 public:
-    /** A read-only random-access iterator to an encoded query parameter.
-    */
+    /** A read-only forward iterator to an encoded query parameter.
+
+        This is a read-only forward iterator to
+        the encoded query parameters. It has the
+        value type @ref query_param and reference
+        type @ref query_param_view.
+
+     */
+#ifdef BOOST_URL_DOCS
+    using iterator = __see_below__;
+#else
     class iterator;
+#endif
 
-    /** A read-only random-access iterator to an encoded query parameter.
+    /** A read-only forward iterator to an encoded query parameter.
```

Comment:
No. If we remove the javadoc from `const_iterator`, it goes from this

![image](https://user-images.githubusercontent.com/5369819/169189960-435c2882-b6c4-487a-b902-4dad31bf20b8.png)

![image](https://user-images.githubusercontent.com/5369819/169190727-49361676-34ee-481e-ace6-88a6693765ee.png)

to this

![image](https://user-images.githubusercontent.com/5369819/169190119-f7ad59fc-8c25-4ca7-b2f4-e95138034608.png)

![image](https://user-images.githubusercontent.com/5369819/169190271-95c26f85-4db6-4cf4-a045-c62d73c7f199.png)

Reply:
This can be a simple

```c++
    /// @copydoc iterator
    using const_iterator = iterator;
```

It renders exactly as before on `iterator`:

![image](https://user-images.githubusercontent.com/324097/169671834-f4a82ca8-aa8f-4191-be88-4ebefad7f6ec.png)

Reply:
`copydoc` works?

Reply:
I posted (local) picture proof!